library/core/src/slice/mod.rs

```diff
@@ -707,7 +705,7 @@ impl<T> [T] {
                 ptr.add(self.len())
             };
 
-            Iter { ptr: NonNull::new_unchecked(ptr as *mut T), end, _marker: marker::PhantomData }
+            Iter::new(NonNull::new_unchecked(ptr as *mut T), end)
         }
```

Comment:
hmm looking at this, would it potentially even make sense for iter new to take `&[T]` and move all of the building into
that function?

This way the `slice` impl block will be below 3000 lines and a bit easier to read. Otherwise this looks good to me.

Reply:
Done. However the iter.rs file is big with 2015 lines.